src/wallet/wallet.cpp

```diff
@@ -804,6 +810,36 @@ void CWallet::AddToSpends(const uint256& wtxid)
         AddToSpends(txin.prevout, wtxid);
 }
 
+bool CWallet::EncryptMnemonic(CKeyingMaterial& master_key)
+{
+    if (!mapCryptedKeys.empty() || IsCrypted())
+        return false;
+
+      auto& meta = mapKeyMetadata[hdChain.masterKeyID];
+      auto& mnemonic = meta.mnemonic;
+
+      CKeyingMaterial secret{mnemonic.begin(), mnemonic.end()};
+
+      auto master_key_id_string = hdChain.masterKeyID.ToString();
+      auto master_key_id_hash = Hash(master_key_id_string.begin(), master_key_id_string.end());
+
+      std::vector<unsigned char> encrypted_mnemonic;
+      if (!EncryptSecret(master_key, secret, master_key_id_hash, encrypted_mnemonic))
+          return false;
+
+      mnemonic.resize(encrypted_mnemonic.size());
+      std::copy(encrypted_mnemonic.begin(), encrypted_mnemonic.end(), mnemonic.begin());
+
+      auto hdkey = mapKeys.find(hdChain.masterKeyID);
+      if (hdkey == mapKeys.end()) {
+          return false;
+      }
+      
+      CWalletDB walletdb{*dbw};
+      walletdb.WriteKeyMetadata(hdkey->second.GetPubKey(), meta);
```

Comment:
`return walletdb.WriteKeyMetadata(hdkey->second.GetPubKey(), meta);` here to check if it was written?